ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/blobs/DataUnavailableBlockPool.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright ConsenSys Software Inc., 2023
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.statetransition.blobs;
+
+import java.time.Duration;
+import java.util.Optional;
+import java.util.Queue;
+import java.util.concurrent.ArrayBlockingQueue;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.async.Cancellable;
+import tech.pegasys.teku.spec.datastructures.blocks.SignedBeaconBlock;
+import tech.pegasys.teku.spec.logic.common.statetransition.results.BlockImportResult;
+import tech.pegasys.teku.statetransition.block.BlockManager;
+
+/**
+ * This pool is designed to track chain tips blocks that has been attempted to import but failed due
+ * to data unavailability. These are chain tips only because blocks with unknown parent will fail
+ * with UNKNOWN_BLOCK and will be tracked in the block pendingPool.
+ */
+public class DataUnavailableBlockPool {
+  private static final Logger LOG = LogManager.getLogger();
+
+  private static final Duration WAIT_BEFORE_RETRY = Duration.ofSeconds(1);
+
+  private final Queue<SignedBeaconBlock> awaitingDataAvailabilityQueue =
```

Comment:
Even so you've added clear class description, I'd add here something like "only fork tips here"